src/main/scala/mesosphere/marathon/io/IO.scala

```diff
@@ -85,6 +85,15 @@ object IO {
     new String(out.toByteArray, "UTF-8")
   }
 
+  def withResource[T](path: String)(fn: InputStream => T): Option[T] = {
```

Comment:
I am not convinced that this code does the proper input path sanitisation. How do we prevent that someone e.g. accesses the configuration files with potential secrets this way?